pkg/acpi/acpi.go

```diff
@@ -62,6 +68,20 @@ func gencsum(b []uint8) uint8 {
 	return ^csum + 1
 }
 
+// getaddr gets an address, be it 64 or 32 bits, at the 64 or 32 bit offset, giving preference
+// to the 64-bit one.
+func getaddr(b []byte, addr64, addr32 int64) (int64, error) {
+	var a64 int64
+	if err := binary.Read(io.NewSectionReader(bytes.NewReader(b), addr64, 8), binary.LittleEndian, &a64); err == nil && a64 != 0 {
```

Comment:
binary.Read(b[addr64:addr64+8], binary.LittleEndian

?

Reply:
This is a nice easy way of avoid all the checking of size you might want to do for ill-formed []byte. It's lazy but it works ...